build-logic/src/main/java/qsl/internal/dependency/QslModuleDependency.java

```diff
@@ -0,0 +1,58 @@
+package qsl.internal.dependency;
+
+import javax.inject.Inject;
+import org.gradle.api.Named;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.provider.Property;
+import org.jetbrains.annotations.NotNull;
+
+public class QslModuleDependency implements Named {
+	private final String name;
+	private final Property<ConfigurationType> configuration;
+
+	@Inject
+	public QslModuleDependency(String name, ObjectFactory objects) {
+		this.name = name;
+
+		configuration = objects.property(ConfigurationType.class);
+		configuration.convention(ConfigurationType.API);
+	}
+
+	public Property<ConfigurationType> getConfiguration() {
+		return configuration;
+	}
+
+	public void testmod() {
```

Comment:
this should be testmodOnly() so that it's more obvious this dependency isn't included in the main pom